api/v1alpha1/virtualmachine_types.go

```diff
@@ -185,6 +189,26 @@ type TCPSocketAction struct {
 	Host string `json:"host,omitempty"`
 }
 
+// The guest heartbeat status.
+type GuestHeartbeatStatus string
+
+// See govmomi.vim25.types.ManagedEntityStatus
+const (
+	// Intermittent heartbeat. May be due to guest load.
+	YellowStatus GuestHeartbeatStatus = "yellow"
+	// Guest operating system is responding normally.
+	GreenStatus  GuestHeartbeatStatus = "green"
+)
+
+// GuestHeartbeatAction describes an action based on the guest heartbeat.
+type GuestHeartbeatAction struct {
```

Comment:
The name `GuestHeartbeatAction` feels like we will take certain actions based on the heartbeat of the Guest. But we are not doing that here. Maybe we can call it `GuestHeartbeatProbe` since this behaves more like a probe?

Reply:
`Action` is based on the existing `TcpSocketAction` name. I'm not really a fan of that name but think we should keep consistency.

Reply:
Makes sense to me.